app/audit/DataEventFactory.scala

```diff
@@ -74,9 +74,9 @@ object DataEventFactory {
 
     val detail = Json.obj(
       "totalDebt" -> journey.debits.map(_.amount).sum.toString,
-      "spending" -> journey.maybeSpending.map(_.totalSpending).getOrElse(BigDecimal(0)).toString,
-      "income" -> journey.maybeIncome.map(_.totalIncome).getOrElse(BigDecimal(0)).toString,
       "halfDisposableIncome" -> (journey.remainingIncomeAfterSpending / 2).toString,
+      "income" -> Json.toJson(AuditIncome.fromIncome(journey.maybeIncome)),
```

Comment:
we've got 
```scala
final case class Income(budgetLines: IncomeBudgetLine*)

final case class IncomeBudgetLine(category: IncomeCategory, amount: BigDecimal)

sealed trait IncomeCategory extends EnumEntry with EnumEntry.Uncapitalised
```
If we wanted to make this more future-proof we could perhaps derive a `JsObject` directly from `Income` instead of creating an `AuditIncome` case class which we would need to update if anything was added or removed from `Income` in the future. You may be able to do something like:
```scala
def toJson(income: Income): JsObject =  {
  val map = income.budgetLines.map{ budgetLine =>
    budgetLine.category.entryName -> JsNumber(budgetLine.amount)
  }.toMap
  JsObject(map)
}
```
(may or may not work - I'm coding in github here 😆).  You'd be able to do a similar thing for outgoings perhaps.

Reply:
All done 😮‍💨